### ISSUE TITLE
Missing `tableauhyperapi.h` during pip install

### ISSUE THREAD

**Issue by akosfurton** [innobi/pantab#40]
When performing a clean install of pantab from pip (pip install tableauhyperapi), the extension module breaks.

Full traceback below:

```
$ pip install pantab
Collecting pantab
  Using cached https://files.pythonhosted.org/packages/e3/82/e3844ce038e52f9927f43f5ab94984f9bbfb6f6dee25ea1fbd6e4b40eeae/pantab-0.1.0.tar.gz
Requirement already satisfied: pandas in /home/akos/akos_venv/lib/python3.6/site-packages (from pantab) (0.25.3)
Requirement already satisfied: numpy>=1.13.3 in /home/akos/akos_venv/lib/python3.6/site-packages (from pandas->pantab) (1.17.4)
Requirement already satisfied: python-dateutil>=2.6.1 in /home/akos/akos_venv/lib/python3.6/site-packages (from pandas->pantab) (2.8.1)
Requirement already satisfied: pytz>=2017.2 in /home/akos/akos_venv/lib/python3.6/site-packages (from pandas->pantab) (2019.3)
Requirement already satisfied: six>=1.5 in /home/akos/akos_venv/lib/python3.6/site-packages (from python-dateutil>=2.6.1->pandas->pantab) (1.13.0)
Building wheels for collected packages: pantab
  Building wheel for pantab (setup.py) ... error
  ERROR: Command errored out with exit status 1:
   command: /home/akos/akos_venv/bin/python3.6 -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-p1jbd85j/pantab/setup.py'"'"'; __file__='"'"'/tmp/pip-install-p1jbd85j/pantab/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' bdist_wheel -d /tmp/pip-wheel-o3f3iwl1 --python-tag cp36
       cwd: /tmp/pip-install-p1jbd85j/pantab/
  Complete output (28 lines):
  running bdist_wheel
  running build
  running build_py
  creating build
  creating build/lib.linux-x86_64-3.6
  creating build/lib.linux-x86_64-3.6/benchmarks
  copying benchmarks/__init__.py -> build/lib.linux-x86_64-3.6/benchmarks
  copying benchmarks/custom.py -> build/lib.linux-x86_64-3.6/benchmarks
  copying benchmarks/benchmarks.py -> build/lib.linux-x86_64-3.6/benchmarks
  creating build/lib.linux-x86_64-3.6/pantab
  copying pantab/__init__.py -> build/lib.linux-x86_64-3.6/pantab
  copying pantab/_writer.py -> build/lib.linux-x86_64-3.6/pantab
  copying pantab/_reader.py -> build/lib.linux-x86_64-3.6/pantab
  copying pantab/_types.py -> build/lib.linux-x86_64-3.6/pantab
  creating build/lib.linux-x86_64-3.6/pantab/tests
  copying pantab/tests/__init__.py -> build/lib.linux-x86_64-3.6/pantab/tests
  copying pantab/tests/test_writer.py -> build/lib.linux-x86_64-3.6/pantab/tests
  copying pantab/tests/conftest.py -> build/lib.linux-x86_64-3.6/pantab/tests
  copying pantab/tests/test_roundtrip.py -> build/lib.linux-x86_64-3.6/pantab/tests
  copying pantab/tests/test_reader.py -> build/lib.linux-x86_64-3.6/pantab/tests
  running build_ext
  building 'libwriter' extension
  creating build/temp.linux-x86_64-3.6
  creating build/temp.linux-x86_64-3.6/pantab
  gcc -pthread -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -fPIC -I/home/akos/akos_venv/include -I/usr/local/include/python3.6m -c pantab/_writermodule.c -o build/temp.linux-x86_64-3.6/pantab/_writermodule.o
  pantab/_writermodule.c:6:29: fatal error: tableauhyperapi.h: No such file or directory
  compilation terminated.
  error: command 'gcc' failed with exit status 1
  ----------------------------------------
  ERROR: Failed building wheel for pantab```

**Comment by WillAyd** [comment]
Can you share what platform this is on and which tableau hyper api Python Bindung you have installed?

Unfortunately not yet possible to provide a manylinux wheel for Linux so this is trying to build from source on install. Something must be going awry

**Comment by akosfurton** [comment]
The platform is `Ubuntu 16.04.6 LTS (GNU/Linux 4.15.0-1063-azure x86_64)` and I have the tableau API from `http://downloads.tableau.com/tssoftware/tableauhyperapi-0.0.8953-py3-none-linux_x86_64.whl` installed in my virtual environment. The version of Python is 3.6.9

I have been using the `0.0.1.b5` version for a few weeks with no issues.

From what I can gather of the wheel logs, it may be that the tableauhyperapi.h file was not copied into the build folder

**Comment by WillAyd** [comment]
The latest release uses a C extension to boost performance and somewhat unfortunately for Linux users I can’t distribute a manylinux wheel. I think the tableauhyperapi has some manylinux incompatibilities 

If you clone this directory and try to pip install from source does it work?

**Comment by WillAyd** [comment]
Let me know if you can confirm. If from source works I think confirms its just something missing in setup.py that needs to be updated

**Comment by akosfurton** [comment]
hmm... when I clone and install from source, it seems that setup.py cannot find the proper location of the tableauhyperapi package. 

```(akos_venv) akos@AMRMP2330406802L:~/sunrise_promo$ pip install git+https://github.com/innobi/pantab
Collecting git+https://github.com/innobi/pantab
  Cloning https://github.com/innobi/pantab to /tmp/pip-req-build-mpab23yj
  Running command git clone -q https://github.com/innobi/pantab /tmp/pip-req-build-mpab23yj
  Installing build dependencies ... done
  Getting requirements to build wheel ... error
  ERROR: Command errored out with exit status 1:
   command: /home/akos/akos_venv/bin/python3.6 /home/akos/akos_venv/lib/python3.6/site-packages/pip/_vendor/pep517/_in_process.py get_requires_for_build_wheel /tmp/tmpcsmylbyv
       cwd: /tmp/pip-req-build-mpab23yj
  Complete output (18 lines):
  Traceback (most recent call last):
    File "/home/akos/akos_venv/lib/python3.6/site-packages/pip/_vendor/pep517/_in_process.py", line 257, in <module>
      main()
    File "/home/akos/akos_venv/lib/python3.6/site-packages/pip/_vendor/pep517/_in_process.py", line 240, in main
      json_out['return_val'] = hook(**hook_input['kwargs'])
    File "/home/akos/akos_venv/lib/python3.6/site-packages/pip/_vendor/pep517/_in_process.py", line 91, in get_requires_for_build_wheel
      return hook(config_settings)
    File "/tmp/pip-build-env-ffgh2ql0/overlay/lib/python3.6/site-packages/setuptools/build_meta.py", line 146, in get_requires_for_build_wheel
      return self._get_build_requires(config_settings, requirements=['wheel'])
    File "/tmp/pip-build-env-ffgh2ql0/overlay/lib/python3.6/site-packages/setuptools/build_meta.py", line 127, in _get_build_requires
      self.run_setup()
    File "/tmp/pip-build-env-ffgh2ql0/overlay/lib/python3.6/site-packages/setuptools/build_meta.py", line 237, in run_setup
      self).run_setup(setup_script=setup_script)
    File "/tmp/pip-build-env-ffgh2ql0/overlay/lib/python3.6/site-packages/setuptools/build_meta.py", line 142, in run_setup
      exec(compile(code, __file__, 'exec'), locals())
    File "setup.py", line 5, in <module>
      from tableauhyperapi.impl.util import find_hyper_api_dll
  ModuleNotFoundError: No module named 'tableauhyperapi'
  ----------------------------------------
ERROR: Command errored out with exit status 1: /home/akos/akos_venv/bin/python3.6 /home/akos/akos_venv/lib/python3.6/site-packages/pip/_vendor/pep517/_in_process.py get_requires_for_build_wheel /tmp/tmpcsmylbyv Check the logs for full command output.
(akos_venv) akos@AMRMP2330406802L:~/$ which tableauhyperapi
(akos_venv) akos@AMRMP2330406802L:~/$ python
Python 3.6.9 (default, Oct  7 2019, 19:57:21)
[GCC 5.4.0 20160609] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> from tableauhyperapi.impl.util import find_hyper_api_dll
>>> find_hyper_api_dll()
PosixPath('/home/akos/akos_venv/lib/python3.6/site-packages/tableauhyperapi/bin/libtableauhyperapi.so')
>>>```

**Comment by WillAyd** [comment]
Did you run “pip install <repo>” above? If so can you alternately download the repo to your file system and try an editable install
from the project root?

https://stackoverflow.com/a/20043907

**Comment by chillerno1** [comment]
Did some investigation of this issue and found "tableauhyperapi.h" wasn't being copied when sdist was run, and therefore pip install pantab was throwing a missing file error: "pantab/_writermodule.c:6:29: fatal error: tableauhyperapi.h: No such file or directory". 

Came across this comment: https://github.com/pypa/packaging-problems/issues/84#issuecomment-384520493 which suggested updating MANIFEST.in to include "recursive-include pantab *.h". After adding this I can see sdist now picks up the missing header:

```
...
copying pantab/_types.py -> pantab-0.1.0/pantab                                                                                         
copying pantab/_writer.py -> pantab-0.1.0/pantab                                                                                          
copying pantab/_writermodule.c -> pantab-0.1.0/pantab                                                                              
copying pantab/tableauhyperapi.h -> pantab-0.1.0/pantab           
...
```

Install runs correctly after rebuild: 

```
...
Successfully built pantab
Installing collected packages: six, python-dateutil, pytz, numpy, pandas, pantab
Successfully installed numpy-1.17.4 pandas-0.25.3 pantab-0.1.0 python-dateutil-2.8.1 pytz-2019.3 six-1.13.0
```

**Comment by akosfurton** [comment]
@chillerno1 's solution worked for me as well. Thank you!

**Comment by WillAyd** [comment]
This should be resolved in pantab 0.1.1 which is now available on PyPi

Thanks again @chillerno1 for the investigation and fix!
